skimage/exposure/_adapthist.py

```diff
@@ -116,74 +117,104 @@ def _clahe(image, kernel_size, clip_limit, nbins):
     minimum and maximum value as the input image. A clip limit smaller than 1
     results in standard (non-contrast limited) AHE.
     """
-    row_step = int(image.shape[0] // np.ceil(image.shape[0] / kernel_size[0]))
-    col_step = int(image.shape[1] // np.ceil(image.shape[1] / kernel_size[1]))
+    # pad the image so that the shape in each dimension
+    # is a multiple of the relevant kernel_size
+    pad_end_per_dim = [0] * image.ndim
+    for dim in range(image.ndim):
+        while (image.shape[dim] + pad_end_per_dim[dim]) % kernel_size[dim]:
```

Comment:
you can compute this without the while loop I think: 
```
pad_end_per_dim[dim] = (kernel_size[dim] - image.shape[dim] % kernel_size[dim]) %kernel_size[dim]
```
not tested but it should work :-)

Reply:
You're right of course!